ui/src/app/views/workspace-library.tsx

```diff
@@ -0,0 +1,176 @@
+  constructor(props) {
+    super(props);
+    this.state = {
+      currentTab: libraryTabEnums.PUBLISHED_WORKSPACES,
+      errorText: '',
+      workspaceList: [],
+      workspacesLoading: true
+    };
+  }
+
+  componentDidMount() {
+    this.reloadPublishedWorkspaces();
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    // Reload libraries when switching tabs
+    if (this.state.currentTab !== prevState.currentTab) {
+      console.log('here');
+      this.updateListedWorkspaces(this.state.currentTab);
+    }
+  }
+
+  updateListedWorkspaces(tabEnum) {
+    switch (tabEnum) {
+      case tabEnum === libraryTabEnums.PUBLISHED_WORKSPACES:
+        this.reloadPublishedWorkspaces();
+        break;
+      default:
+        break;
+    }
+  }
+
+  async reloadPublishedWorkspaces() {
+    this.setState({workspacesLoading: true});
+    try {
+      const workspacesReceived = await workspacesApi().getPublishedWorkspaces();
+      workspacesReceived.items.sort(
+        (a, b) => a.workspace.name.localeCompare(b.workspace.name));
+      this.setState({workspaceList: workspacesReceived.items
+          .map(w => new WorkspacePermissions(w))});
+      this.setState({workspacesLoading: false});
+    } catch (e) {
+      const response = ErrorHandlingService.convertAPIError(e) as unknown as ErrorResponse;
+      this.setState({errorText: response.message});
+    }
+  }
+
+  render() {
+    const {profile} = this.props.profileState;
+    const {currentTab, workspaceList, workspacesLoading} = this.state;
+    return <div style={{display: 'flex', flexDirection: 'row', height: '100%'}}>
+      <div style={styles.navPanel}>
+        <div style={{display: 'flex', flexDirection: 'column'}}>
+          {libraryTabs.map((tab, i) => {
+            return <React.Fragment key={i}>
+              <LibraryTab icon={tab.icon} title={tab.title} selected={currentTab === tab}
+                          onClick={() => this.setState({currentTab: tab})}/>
+                {i !== libraryTabs.length - 1 &&
+                <hr style={{width: '100%', margin: '0.5rem 0'}}/>}
+            </React.Fragment>;
+          })}
+        </div>
+      </div>
+      <div style={{padding: '1rem', width: '100%'}}>
+        <Header>RESEARCHER WORKBENCH WORKSPACE LIBRARY</Header>
+        <div style={{color: colors.primary, fontSize: 16, marginTop: '1rem'}}>
+          Search through featured and public workspaces.
+        </div>
+        <div style={{display: 'flex', flexDirection: 'column', marginTop: '2rem'}}>
+          <div style={{display: 'flex', flexDirection: 'row'}}>
+            <ClrIcon shape={currentTab.icon} style={styles.iconStyling}
+                     class='is-solid' size={24}/>
+            <div style={{color: colors.primary, fontSize: 18, fontWeight: 600}}>
+              {currentTab.title}
+            </div>
+          </div>
+          <hr style={styles.divider}/>
+          {currentTab === libraryTabEnums.PUBLISHED_WORKSPACES &&
```

Comment:
What does it show if the currentTab is FEATURED_WORKSPACES? I would either implement that, or remove the tab option for now and just have the single tab.

Reply:
it shows nothing for now, but i wanted to build that functionality so that the next ticket is trivial.  would opt for keeping the logic